openslides_backend/presenter/search_users_by_name_email.py

```diff
@@ -0,0 +1,127 @@
+                        "email": {"type": "string"},
+                    },
+                },
+            },
+        },
+        "required": ["permission_type", "permission_id", "search"],
+        "additionalProperties": False,
+    }
+)
+
+
+@register_presenter("search_users_by_name_email")
+class SearchUsersByNameEmail(BasePresenter):
+    """
+    Collects users with exect usernames or exact emails.
+    """
+
+    schema = search_users_by_name_email_schema
+
+    def get_result(self) -> Any:
+        self.check_permissions(self.data["permission_type"], self.data["permission_id"])
+        result: Dict[str, List[Dict[str, Union[str, int]]]] = {}
+        for search in self.data["search"]:
+            username = search.get("username", "")
+            email = search.get("email", "")
+            if username and email:
+                filter_ = Or(
+                    FilterOperator("username", "=", username),
+                    FilterOperator("email", "=", email),
+                )
+            elif username:
+                filter_ = FilterOperator("username", "=", username)
+            elif email:
+                filter_ = FilterOperator("email", "=", email)
+            else:
+                continue
+            instances = self.datastore.filter(
+                Collection("user"), filter_, ["id", "first_name", "last_name", "email"]
+            )
+
+            result[f"{username}/{email}"] = [
+                instance for instance in instances.values()
+            ]
+        return result
+
+    def check_permissions(self, permission_type: int, permission_id: int) -> None:
+        if has_organization_management_level(
+            self.datastore, self.user_id, OrganizationManagementLevel.CAN_MANAGE_USERS
+        ):
+            return
+        if permission_type == UserScope.Organization.value:
+            raise MissingPermission(OrganizationManagementLevel.CAN_MANAGE_USERS)
+        if permission_type == UserScope.Committee.value:
+            if has_committee_management_level(
+                self.datastore,
+                self.user_id,
+                CommitteeManagementLevel.CAN_MANAGE,
+                permission_id,
+            ):
+                return
+            else:
+                raise MissingPermission(
+                    {CommitteeManagementLevel.CAN_MANAGE: permission_id}
+                )
+        if has_perm(
+            self.datastore,
+            self.user_id,
+            Permissions.User.CAN_MANAGE,
+            permission_id,
+        ):
+            return
+        else:
+            meeting = self.datastore.get(
+                FullQualifiedId(Collection("meeting"), permission_id),
+                ["committee_id"],
+            )
+            if has_committee_management_level(
+                self.datastore,
+                self.user_id,
+                CommitteeManagementLevel.CAN_MANAGE,
+                meeting.get("committee_id", 0),
```

Comment:
I would prefer to use `[]` here since the `committee_id` is required and should always be present. If it is not, it is suitable to throw a 500, so the user and eventually we notice it and can debug and hopefully fix it. By providing `0` to `has_committee_management_level`, no error will be thrown because it will try to access the field `committee_$0_management_level` on the user which will always be `None`. This will probably go unnoticed or may even lead to hard-to-debug bugs.

Reply:
This is made because of mypy-check. `has_committee_management_level` expects an `int`-type. [] is not allowed, because it should be `int`. Changed it to `cast`.

Reply:
if you do `cast(int, meeting["committee_id"])` this should either work if committee id is correctly filled or throw an attribute error if it is empty. As I said, I prefer the 500 here since we want the use to notice if the database is corrupt. Or am I missing some mypy caveat here?

Reply:
I will check it and throw an exception. On casting, I tried it, the error message is very abstruse (committee/None is not a valid FQID).

Reply:
But if you use the index operator you will never get to this point because you will get an attribute error before you even cast :D that's what I wrote above. But anyway, this PR does not have to wait because of such a detail

Reply:
Yes, you are right with the index operator. But now we have decided message:D